roles/openshift_health_checker/openshift_checks/docker_image_availability.py

```diff
@@ -61,10 +61,15 @@ def __init__(self, *args, **kwargs):
         # for the oreg_url registry there may be credentials specified
         components = self.get_var("oreg_url", default="").split('/')
         self.registries["oreg"] = "" if len(components) < 3 else components[0]
+
+        # Retrieve and template registry credentials, if provided
         self.skopeo_command_creds = ""
         oreg_auth_user = self.get_var('oreg_auth_user', default='')
         oreg_auth_password = self.get_var('oreg_auth_password', default='')
         if oreg_auth_user != '' and oreg_auth_password != '':
+            if self._templar is not None:
+                oreg_auth_user = self._templar.template(oreg_auth_user)
+                oreg_auth_password = self._templar.template(oreg_auth_password)
```

Comment:
@michaelgugino Actually now that I think of it, this has the small disadvantage that if your user/pass happens to be `{{foo}}` then you will probably get an exception at this point and the whole thing will be rather confusing. I hate that there's essentially no way to distinguish between "the user wanted to provide their password in a var from somewhere else" and "the user wanted a password with double braces in it". Granted the former case is a lot more likely than the latter, so maybe this is the best we can do. Or can we? I just hate edge cases like this.

Reply:
If someone sets their password to "{{ foo }}" ansible is always going to try to template that.  We're not doing anything abnormal here, this is exactly how the built-in action plugins template variables.

The user would need to escape those characters to supply such a value for any variable.  This problem is not unique to this module, thus I don't think we should go out of our way to handle it.

Reply:
>  I hate that there's essentially no way to distinguish between "the user wanted to provide their password in a var from somewhere else" and "the user wanted a password with double braces in it".

Agreed but I think it's reasonable. It should probably be documented just in case.

Reply:
No, good point, any other task would do the same with its vars. I guess get_var should just template everything then. Simple enough...